sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,630 @@
+    // ERROR_WINHTTP_RESPONSE_DRAIN_OVERFLOW
+    // ERROR_WINHTTP_SECURE_FAILURE
+    // ERROR_WINHTTP_SHUTDOWN
+    // ERROR_WINHTTP_TIMEOUT
+    // ERROR_WINHTTP_UNRECOGNIZED_SCHEME
+    // ERROR_NOT_ENOUGH_MEMORY
+    // ERROR_INVALID_PARAMETER
+    // ERROR_WINHTTP_RESEND_REQUEST
+    GetErrorAndThrow("Error while sending a request.");
+  }
+
+  // Chunked transfer encoding is not supported and the content length needs to be known up front.
+  if (streamLength == -1)
+  {
+    throw Azure::Core::Http::TransportException(
+        "When uploading data, the body stream must have a known length.");
+  }
+
+  if (streamLength > 0)
+  {
+    Upload(handleManager);
+  }
+}
+
+void WinHttpTransport::ReceiveResponse(std::unique_ptr<Details::HandleManager>& handleManager)
+{
+  // Wait to receive the response to the HTTP request initiated by WinHttpSendRequest.
+  // When WinHttpReceiveResponse completes successfully, the status code and response headers have
+  // been received.
+  if (!WinHttpReceiveResponse(handleManager->m_requestHandle, NULL))
+  {
+    // Errors include:
+    // ERROR_WINHTTP_CANNOT_CONNECT
+    // ERROR_WINHTTP_CHUNKED_ENCODING_HEADER_SIZE_OVERFLOW
+    // ERROR_WINHTTP_CLIENT_AUTH_CERT_NEEDED
+    // ...
+    // ERROR_WINHTTP_TIMEOUT
+    // ERROR_WINHTTP_UNRECOGNIZED_SCHEME
+    // ERROR_NOT_ENOUGH_MEMORY
+    GetErrorAndThrow("Error while receiving a response.");
+  }
+}
+
+int64_t WinHttpTransport::GetContentLength(
+    std::unique_ptr<Details::HandleManager>& handleManager,
+    HttpMethod requestMethod,
+    HttpStatusCode responseStatusCode)
+{
+  DWORD dwContentLength = 0;
+  DWORD dwSize = sizeof(dwContentLength);
+
+  // For Head request, set the length of body response to 0.
+  // Response will give us content-length as if we were not doing Head saying what would be the
+  // length of the body. However, server won't send any body.
+  // For NoContent status code, also need to set contentLength to 0.
+  int64_t contentLength = 0;
+
+  // Get the content length as a number.
+  if (requestMethod != HttpMethod::Head && responseStatusCode != HttpStatusCode::NoContent)
+  {
+    if (!WinHttpQueryHeaders(
+            handleManager->m_requestHandle,
+            WINHTTP_QUERY_CONTENT_LENGTH | WINHTTP_QUERY_FLAG_NUMBER,
+            WINHTTP_HEADER_NAME_BY_INDEX,
+            &dwContentLength,
+            &dwSize,
+            WINHTTP_NO_HEADER_INDEX))
+    {
+      contentLength = -1;
+    }
+    else
+    {
+      contentLength = static_cast<int64_t>(dwContentLength);
+    }
+  }
+
+  return contentLength;
+}
+
+std::unique_ptr<Details::WinHttpStream> WinHttpTransport::GetBodyStream(
+    std::unique_ptr<Details::HandleManager>& handleManager,
+    int64_t contentLength)
 {
-  void(context);
-  void(request);
+  return std::make_unique<Details::WinHttpStream>(handleManager, contentLength);
+}
+
+std::unique_ptr<RawResponse> WinHttpTransport::GetRawResponse(
+    std::unique_ptr<Details::HandleManager>& handleManager,
```

Comment:
```suggestion
    std::unique_ptr<Details::HandleManager> handleManager,
```
Explained why this change in the method definition in the header